src/component/resource/file/CreateFileMetadata.tsx

```diff
@@ -1,59 +1,92 @@
 import * as React from "react";
 import {injectIntl} from "react-intl";
-import {
-    CreateResourceMetadata,
-    CreateResourceMetadataProps,
-    CreateResourceMetadataState
-} from "../CreateResourceMetadata";
-import VocabularyUtils from "../../../util/VocabularyUtils";
-import withI18n from "../../hoc/withI18n";
-import {Form} from "reactstrap";
+import withI18n, {HasI18n} from "../../hoc/withI18n";
+import {Button, ButtonToolbar, Col, Form, Row} from "reactstrap";
 import UploadFile from "./UploadFile";
 import TermItFile from "../../../model/File";
 import Resource from "../../../model/Resource";
+import CustomInput from "../../misc/CustomInput";
+import ShowAdvanceAssetFields from "../../asset/ShowAdvancedAssetFields";
+import {AbstractCreateAssetState} from "../../asset/AbstractCreateAsset";
+import {AssetData} from "../../../model/Asset";
 
-interface CreateFileMetadataProps extends CreateResourceMetadataProps {
+interface CreateFileMetadataProps extends HasI18n {
     onCreate: (termItFile : Resource, file: File) => any;
+    onCancel: () => void;
 }
 
-interface CreateFileMetadataState extends CreateResourceMetadataState {
+interface CreateFileMetadataState extends AbstractCreateAssetState, AssetData {
+    iri: string;
+    label: string;
+    generateIri: boolean;
     file?: File;
     dragActive: boolean;
 }
 
-export class CreateFileMetadata extends CreateResourceMetadata<CreateFileMetadataProps, CreateFileMetadataState> {
+export class CreateFileMetadata extends React.Component<CreateFileMetadataProps, CreateFileMetadataState> {
 
     constructor(props: CreateFileMetadataProps) {
         super(props);
         this.state = {
             iri: "",
             label: "",
-            description: "",
-            types: VocabularyUtils.RESOURCE,
             generateIri: true,
```

Comment:
I don't see IRI generation being used in this component any more. So why is this state variable still here?